src/librustc/middle/mem_categorization.rs

```diff
@@ -465,9 +465,11 @@ impl<'a, 'tcx> MemCategorizationContext<'a, 'tcx> {
     ) -> bool {
         self.infcx.map(|infcx| infcx.type_is_copy_modulo_regions(param_env, ty, span))
             .or_else(|| {
-                self.tcx.lift_to_global(&(param_env, ty)).map(|(param_env, ty)| {
-                    ty.is_copy_modulo_regions(self.tcx.global_tcx(), param_env, span)
-                })
+                if (param_env, ty).has_local_value() {
+                    None
```

Comment:
Hmm, if `self.infcx` is `None`, this case should be impossible - it's plausible this code is unnecessarily defensive and `lift_to_global`'s result could've been `unwrap`'d all along.

Reply:
I agree.